include/oneapi/dpl/pstl/utils.h

```diff
@@ -261,6 +264,22 @@ class __not_equal_value
     }
 };
 
+template <typename _Tp>
+class __set_value
+{
+    const _Tp _M_value;
+
+  public:
+    explicit __set_value(const _Tp& __value) : _M_value(__value) {}
+
+    template <typename _Arg>
+    void
+    operator()(_Arg&& __arg) const
+    {
+        std::forward<_Arg>(__arg) = _M_value;
+    }
```

Comment:
```suggestion
    operator()(_Arg& __arg) const
    {
        __arg = _M_value;
    }
```

I do not understand the purpose of a forwarding reference in this context. Moreover, it may lead to assigning to an rvalue.

Reply:
It seems it is for a zip_iterator proxy, which is an rvalue containing references.

Reply:
1. a proxy instance (rvalue) doesn't match to lvalue ref.
2. operator= may be overloaded